App.js

```diff
@@ -26,29 +20,39 @@ const storeData = async (value) => {
 
 const getData = async () => {
 	  try {
-	const jsonValue = await AsyncStorage.getItem('@tasks')
-	 return jsonValue != null ? JSON.parse(jsonValue) : null;
-        } catch(err) {
-		console.log(err);
+      const jsonValue = await AsyncStorage.getItem('@tasks')
+      
+      console.log('inside get promise', jsonValue, JSON.parse(jsonValue))
+      return jsonValue != null ? JSON.parse(jsonValue) : null;
+    } catch(err) {
+      console.log(err);
 	  }
-	}
-			    
+}
+  
+ const addTodo = () => {
+  let id = todos[todos.length - 1]?.id || 0 ;
+   let thisId = id * 1 + 1 + "";
+   const updatedToDoArray = [...todos, { id: thisId, text }];
+  setTodos(updatedToDoArray);
+  storeData(updatedToDoArray)
+  onChangeText("");
+ }
+ 
 
-const remove = id => { 
-setTodos(() =>todos.filter(item => item.id !== id )) ; 
+  const remove = id => { 
+    const updatedToDoArray = todos.filter(item => item.id !== id);
+    setTodos(updatedToDoArray);
 }
 
-useEffect(() => {
- storeData(todos);
-}, [todos] 
-)
+useEffect(() => { 
+  getData().then(storedData => { 
```

Comment:
This is super important! 
`getData` is a Promise. It doesn't return a value like a regular JS function, you can't do 

`const something = getData()`

I mean, you can... but it won't work. You have to use `.then()` with Promises.
You can read more about Promises here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise